### ISSUE TITLE
search for class name in var types

### ISSUE THREAD

**Issue by Trigun87** [icsharpcode/ILSpy#2146]
i think is not possible to search all class that have a single object as a var
something like 
```
class ABC { }
class foo {
var xyz :  ABC
fun ABC xy() {return xyz}
}
```
and by searching ABC u get the result xyz and xy() from foo class

**Comment by siegfriedpammer** [comment]
Sorry, I don't get it. What are you trying to say?

**Comment by Trigun87** [comment]
![image](https://user-images.githubusercontent.com/33032691/92264032-de1f5580-eedd-11ea-99c8-446f756954b4.png)
search for the type of var or the return type of a function , if i have object X and want see where it can be stored i can only use the analyze, but is used by not returned 
the screen i put was bad but i'm on remote and got the first thing i saw :-D the point is the same for ex search for the object RectTransform and get as result classname.locketcontent , etc

**Comment by siegfriedpammer** [comment]
So you are looking for the "Exposed By" analyzer?

**Comment by Trigun87** [comment]
don't know, exposed by is always empty :-)

**Comment by siegfriedpammer** [comment]
Can you provide the assembly for me to check? Thank you!

**Comment by Trigun87** [comment]
the dll?

**Comment by siegfriedpammer** [comment]
yes, please. You can send it to siegfried.pammer@gmail.com, thanks!

**Comment by siegfriedpammer** [comment]
![image](https://user-images.githubusercontent.com/344204/92273370-812a9c00-eeeb-11ea-9c85-10b01cf77421.png)

The uses of f5 in f6 are shown. What am I missing?

**Comment by Trigun87** [comment]
u are right.... but in someclasses i get 0 results... possible?
for ex TradeCreateControl have 0 esposed by, only 1 used by from external class and in that class there isn't a parameter, but he get data from somewhere, didn't checked well but i don't think is a global var

________________________________
Da: Siegfried Pammer <notifications@github.com>
Inviato: venerdì 4 settembre 2020 20:17
A: icsharpcode/ILSpy <ILSpy@noreply.github.com>
Cc: Trigun87 <attila1919@hotmail.com>; Author <author@noreply.github.com>
Oggetto: Re: [icsharpcode/ILSpy] search for class name in var types (#2146)


[image]<https://user-images.githubusercontent.com/344204/92273370-812a9c00-eeeb-11ea-9c85-10b01cf77421.png>

The uses of f5 in f6 are shown. What am I missing?

**Comment by Trigun87** [comment]
i think i found bc sometimes doesn't works...
TradeCreateControl have _item in the parentclass and in the subclasses too, i was analyzing from parent and not from the Step1
no idea if the other times was for this reason too

________________________________
Da: Siegfried Pammer <notifications@github.com>
Inviato: venerdì 4 settembre 2020 20:17
A: icsharpcode/ILSpy <ILSpy@noreply.github.com>
Cc: Trigun87 <attila1919@hotmail.com>; Author <author@noreply.github.com>
Oggetto: Re: [icsharpcode/ILSpy] search for class name in var types (#2146)


[image]<https://user-images.githubusercontent.com/344204/92273370-812a9c00-eeeb-11ea-9c85-10b01cf77421.png>

The uses of f5 in f6 are shown. What am I missing?

**Comment by siegfriedpammer** [comment]
Well, if you find a case where it does not work as expected, please tell me.

**Comment by Trigun87** [comment]
and is possible to do a deep search? like search a string/text in the code?

________________________________
Da: Siegfried Pammer <notifications@github.com>
Inviato: venerdì 4 settembre 2020 21:02
A: icsharpcode/ILSpy <ILSpy@noreply.github.com>
Cc: Trigun87 <attila1919@hotmail.com>; Author <author@noreply.github.com>
Oggetto: Re: [icsharpcode/ILSpy] search for class name in var types (#2146)


Well, if you find a case where it does not work as expected, please tell me.

**Comment by siegfriedpammer** [comment]
No, this is currently not possible.

**Comment by Trigun87** [comment]
in the dll i sent u if u go on TradeCreateControl.Step1._item
and analyze -> assigned by
the result can't be clicked

________________________________
Da: Siegfried Pammer <notifications@github.com>
Inviato: venerdì 4 settembre 2020 21:02
A: icsharpcode/ILSpy <ILSpy@noreply.github.com>
Cc: Trigun87 <attila1919@hotmail.com>; Author <author@noreply.github.com>
Oggetto: Re: [icsharpcode/ILSpy] search for class name in var types (#2146)


Well, if you find a case where it does not work as expected, please tell me.

**Comment by siegfriedpammer** [comment]
This works on my machine. Did you change / reload the assembly list after analyzing the `Step1._item` field? This might invalidate references and cause the tree view to stop working.

**Comment by Trigun87** [comment]
no i didn't change/reload anything if isn't something automatic

btw on the exposed by it doesn't show the object inside of arrays/list/dictionary
isn't a direct relation but is still "exposed by"
________________________________
Da: Siegfried Pammer <notifications@github.com>
Inviato: sabato 5 settembre 2020 17:24
A: icsharpcode/ILSpy <ILSpy@noreply.github.com>
Cc: Trigun87 <attila1919@hotmail.com>; Author <author@noreply.github.com>
Oggetto: Re: [icsharpcode/ILSpy] search for class name in var types (#2146)


This works on my machine. Did you change / reload the assembly list after analyzing the Step1._item field? This might invalidate references and cause the tree view to stop working.

**Comment by siegfriedpammer** [comment]
> btw on the exposed by it doesn't show the object inside of arrays/list/dictionary isn't a direct relation but is still "exposed by"

This is by design, it seems, looking at the original code from 10 years ago. https://github.com/icsharpcode/ILSpy/commit/46e5a2afb98621158ba7b61dcbaf47b78892a37b#diff-b3e48faf7c91381f5618f8c95e437c56R155

@EdHarvey, the original author, might be able to answer this question.

**Comment by Trigun87** [comment]
hello,
i think i found what i was telling u from the start 🙂
[cid:4f83ba14-13b6-4865-b639-aa909e44f2cb]
the b0 var is set only on that function that is never called, i don't think that is possible (ahy set it to 0 and i know that sometimes that var should be >0)
if i'm not wrong deserialize should be a default function for something like json or similar, so it should be used for load the "save", so i can exclude it... the only useful function should be ahb that is never called, but i have a
num3 = b0 / ahs();
and the result is compared to 0,1,2,3,4 but if ahb is never called it should always be 0, since i can see in the game the trigger of the 1,2,3,4 event i don't think that is the case 🙂
https://mega.nz/file/sGxjwK5D#t0-LHlB0LVADeVEVC6A3nbyedaSbwdDe3NCW5I7UT84
[https://mega.nz/rich-file.png]<https://mega.nz/file/sGxjwK5D#t0-LHlB0LVADeVEVC6A3nbyedaSbwdDe3NCW5I7UT84>
2.52 MB file on MEGA<https://mega.nz/file/sGxjwK5D#t0-LHlB0LVADeVEVC6A3nbyedaSbwdDe3NCW5I7UT84>
mega.nz


________________________________
Da: Siegfried Pammer <notifications@github.com>
Inviato: sabato 12 settembre 2020 09:27
A: icsharpcode/ILSpy <ILSpy@noreply.github.com>
Cc: Trigun87 <attila1919@hotmail.com>; Author <author@noreply.github.com>
Oggetto: Re: [icsharpcode/ILSpy] search for class name in var types (#2146)


btw on the exposed by it doesn't show the object inside of arrays/list/dictionary isn't a direct relation but is still "exposed by"

This is by design, it seems, looking at the original code from 10 years ago. 46e5a2a#diff-b3e48faf7c91381f5618f8c95e437c56R155<https://github.com/icsharpcode/ILSpy/commit/46e5a2afb98621158ba7b61dcbaf47b78892a37b#diff-b3e48faf7c91381f5618f8c95e437c56R155>

@EdHarvey<https://github.com/EdHarvey>, the original author, might be able to answer this question.

**Comment by siegfriedpammer** [comment]
Sorry, I do not follow.
